Src/FluentAssertions/Execution/ContinuedAssertionScope.cs

```diff
@@ -12,22 +12,27 @@ namespace FluentAssertions.Execution
     public sealed class ContinuedAssertionScope : IAssertionScope
     {
         private readonly AssertionScope predecessor;
-        private readonly bool predecessorSucceeded;
+        private readonly bool continueAsserting;
 
-        public ContinuedAssertionScope(AssertionScope predecessor, bool predecessorSucceeded)
+        internal ContinuedAssertionScope(AssertionScope predecessor, bool continueAsserting)
         {
-            this.predecessorSucceeded = predecessorSucceeded;
             this.predecessor = predecessor;
+            this.continueAsserting = continueAsserting;
         }
 
         public GivenSelector<T> Given<T>(Func<T> selector)
         {
-            return predecessor.Given(selector);
+            if (continueAsserting)
+            {
+                return predecessor.Given(selector);
+            }
+
+            return new GivenSelector<T>(() => default(T), predecessor, false);
```

Comment:
🙃 Could be shortened to
```suggestion
            return new GivenSelector<T>(() => default, predecessor, false);
```